blaze-core/src/main/scala/org/http4s/blazecore/util/CachingChunkWriter.scala

```diff
@@ -42,37 +42,44 @@ private[http4s] class CachingChunkWriter[F[_]](
   import ChunkWriter._
 
   private[this] var pendingHeaders: StringWriter = _
-  private[this] var bodyBuffer: Buffer[Chunk[Byte]] = Buffer()
+  private[this] val bodyBuffer: Buffer[Chunk[Byte]] = Buffer()
   private[this] var size: Int = 0
 
   override def writeHeaders(headerWriter: StringWriter): Future[Unit] = {
     pendingHeaders = headerWriter
     FutureUnit
   }
 
-  private def addChunk(chunk: Chunk[Byte]): Unit = {
-    bodyBuffer += chunk
-    size += chunk.size
-  }
-
-  private def clear(): Unit = {
+  private def addChunk(chunk: Chunk[Byte]): Unit =
+    if (chunk.nonEmpty) {
+      bodyBuffer += chunk
+      size += chunk.size
+    }
+
+  private def toChunkAndClear: Chunk[Byte] = {
+    val chunk = if (bodyBuffer.isEmpty) {
+      Chunk.empty
+    } else if (bodyBuffer.size == 1) {
+      bodyBuffer.head
```

Comment:
`.headOption` was noticeably slower, so in this case I think `.head` is the right choice.